governance/program/src/instruction.rs

```diff
@@ -0,0 +1,203 @@
+    ///   6. `[writable]` Initialized Signatory account for the issued signatory token
+    ///   7. `[writable]` Initialized Source Token Holding account
+    ///   8. `[]` Source mint account
+    ///   9. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_address]
+    ///   10. '[]` Token program account
+    ///   11. `[]` Rent sysvar
+    InitializeProposal {
+        /// Link to gist explaining proposal
+        description_link: String,
+        /// Name of the proposal
+        name: String,
+    },
+
+    /// [Requires Admin token]
+    /// Adds a signatory to the Proposal which means this Proposal can't leave Draft state until yet another Signatory signs
+    /// As a result of this call the new Signatory will receive a Signatory Token which then can be used to Sign proposal
+    ///
+    ///   0. `[writable]` Initialized Signatory account
+    ///   1. `[writable]` Initialized Signatory Mint account
+    ///   2. `[signer]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_address]
+    ///   6. '[]` Token program account
+    AddSignatory,
+
+    /// [Requires Admin token]
+    /// Removes a Signatory from the Proposal
+    ///
+    ///   0. `[writable]` Signatory account to remove token from
+    ///   1. `[writable]` Signatory Mint account
+    ///   2. `[signer]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. `[signer]` Transfer authority
+    ///   6. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_address]
+    ///   7. '[]` Token program account
+    RemoveSignatory,
+
+    /// [Requires Admin token]
+    /// Adds an instruction to the Proposal. Max of 5 of any  type. More than 5 will throw error
+    ///
+    ///   0. `[writable]` Uninitialized Proposal Instruction account
+    ///   1. `[writable]` Proposal state account
+    ///   2. `[signer]` Admin account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Governance account
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_address]
+    ///   6. `[]` Governance program account
+    AddCustomSingleSignerInstruction {
+        /// Slot waiting time between vote period ending and this being eligible for execution
+        hold_up_time: Slot,
+        /// Instruction
+        instruction: Instruction,
+        /// Position in instruction array
+        position: u8,
+    },
+
+    /// [Requires Admin token]
+    /// Remove instruction from the Proposal
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Proposal instruction account
+    ///   2. `[signer]` Admin account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_address]
+    RemoveInstruction,
```

Comment:
Do we need RemoveInstruction?  Seems like you'd know what you want to in advance, and would only ever add instructions you really want

Reply:
The idea is that while the `Proposal` is in `Draft` mode you can keep editing it until you are happy with it. This is why the Proposal editing instructions exist `AddCustomSingleSignerInstruction`, `RemoveInstruction`, `UpdateInstructionHoldUpTime`. Without that if you make a mistake you would have to delete `Proposal`